lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * API to interact with global settings & styles.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds global style rules to the inline style for each block.
+ */
+function wp_add_global_styles_for_blocks() {
+	$tree = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
+	// TODO some nodes dont have a name...
+	$block_nodes = $tree->get_styles_block_nodes();
+	foreach ( $block_nodes as $metadata ) {
+		$block_css  = $tree->get_styles_for_block( $metadata );
+		$block_name = str_replace( 'core/', '', $metadata['name'] );
+		wp_add_inline_style( 'wp-block-' . $block_name, $block_css );
```

Comment:
Took me a while to figure out that the inline style exists at this stage and this only adds to the `'wp-block-' . $block_name` handler name :) These block inline styles are added on `block_render` and this is how we only get the ones that are actually on page, right?

Reply:
Let's add a comment to that effect here as it took me a while to figure out as well.